model_analyzer/config/generate/quick_run_config_generator.py

```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
     def _get_next_run_config(self) -> RunConfig:
         run_config = RunConfig(self._triton_env)
 
-        for i, _ in enumerate(self._models):
-            mrc = self._get_next_model_run_config(i)
+        model_index = 0
+        for model in self._models:
+            mrc, model_index = self._get_next_model_run_config(
+                model, model_index)
             run_config.add_model_run_config(mrc)
 
         return run_config
 
-    def _get_next_model_run_config(self, model_num: int) -> ModelRunConfig:
-        mc = self._get_next_model_config(model_num)
+    def _get_next_model_run_config(
```

Comment:
this function does a few different things. Can you refactor it?

Reply:
It does two things: creates the next MRC and returns the next model count index. I thought about breaking out the model count index, but it's needed when creating the MRC. I'm not sure how I could refactor this method further.